pkgs/build-support/node/prefetch-npm-deps/src/cacache.rs

```diff
@@ -115,13 +123,27 @@ impl Cache {
             size: data.len(),
             metadata: Metadata {
                 url,
+                req_headers,
                 options: Options { compress: true },
             },
         })?;
 
-        let mut file = File::options().append(true).create(true).open(index_path)?;
-
-        write!(file, "{:x}\t{data}", Sha1::new().chain(&data).finalize())?;
+        let mut file = File::options()
+            .append(true)
+            .create(true)
+            .open(&index_path)?;
+
+        // cacache format uses newline as entry separator (see cacache entry-index.js)
+        // Only add newline prefix if file already has content, to maintain backwards compatibility
```

Comment:
```suggestion
        // Only add newline prefix if file already has content, to maintain backwards compatibility with previous versions of prefetch-npm-deps, which handled this case incorrectly.
```
for posterity of what we’re maintaining back compat with.

(I’m very willing to always write newline prefix for cache ver >= 2, btw, though you’d have to plumb the version through and idk if that’s worth it at this point.)